### PR TITLE
protect against duplicate cidr blocks or sgs in aws_security_group

### PR DESCRIPTION
**Pull request by lukedemi**

### CODE REVIEW

**Comment by heimdall-asguard** [comment]
Reviews      | 0/1 | Message
------------ | ------------- | ----------




[More info](https://heimdall-public.cbhq.net/v1/code_change?repository=coinbase/geoengineer&ref=f14dd91f9491d3ebf139bf820b99c300b33c3d24)

File: lib/geoengineer/resources/aws_security_group.rb [hunk 1]
```diff
@@ -26,6 +27,13 @@ def flatten_cidr_and_sg_blocks
     end
   end
 
+  def uniq_cidr_and_sg_blocks
+    (self.all_ingress + self.all_egress).each do |in_eg|
+      in_eg.cidr_blocks      = in_eg.cidr_blocks.uniq     if in_eg.cidr_blocks
+      in_eg.security_groups  = in_eg.security_groups.uniq if in_eg.security_groups
+    end
+  end
```
Comment (sds) [inline lib/geoengineer/resources/aws_security_group.rb]: Could combine with `flatten_cidr_and_sg_blocks`, but this will do the trick just dandy.

**Comment by heimdall-asguard** [comment]
User failed mfa authentication

**Comment by grahamjenson** [comment]
+1

**Comment by heimdall-asguard** [comment]
Reviews      | 1/1 | Message
------------ | ------------- | ----------
grahamjenson | 1 | +1 



[More info](https://heimdall-public.cbhq.net/v1/code_change?repository=coinbase/geoengineer&ref=f14dd91f9491d3ebf139bf820b99c300b33c3d24)

**Comment by heimdall-asguard** [comment]
Reviews      | 2/1 | Message
------------ | ------------- | ----------
grahamjenson | 1 | +1 
chwevans | 1 | approved 



[More info](https://heimdall-public.cbhq.net/v1/code_change?repository=coinbase/geoengineer&ref=f14dd91f9491d3ebf139bf820b99c300b33c3d24)

**Comment by heimdall-asguard** [comment]
Reviews      | 3/1 | Message
------------ | ------------- | ----------
grahamjenson | 1 | +1 
chwevans | 1 | approved 
sds | 1 | approved 



[More info](https://heimdall-public.cbhq.net/v1/code_change?repository=coinbase/geoengineer&ref=f14dd91f9491d3ebf139bf820b99c300b33c3d24)

**Comment by grahamjenson** [comment]
+1

**Comment by heimdall-asguard** [comment]
Reviews      | 3/1 | Message
------------ | ------------- | ----------
grahamjenson | 1 | +1
 
chwevans | 1 | approved 
sds | 1 | approved 



[More info](https://heimdall-public.cbhq.net/v1/code_change?repository=coinbase/geoengineer&ref=f14dd91f9491d3ebf139bf820b99c300b33c3d24)

**Comment by grahamjenson** [comment]
+1

**Comment by heimdall-asguard** [comment]
Reviews      | 3/1 | Message
------------ | ------------- | ----------
grahamjenson | 1 | +1 
chwevans | 1 | approved 
sds | 1 | approved 



[More info](https://heimdall-public.cbhq.net/v1/code_change?repository=coinbase/geoengineer&ref=f14dd91f9491d3ebf139bf820b99c300b33c3d24)

**Comment by grahamjenson** [comment]
+1

**Comment by heimdall-asguard** [comment]
Reviews      | 3/1 | Message
------------ | ------------- | ----------
grahamjenson | 1 | +1 
chwevans | 1 | approved 
sds | 1 | approved 



[More info](https://heimdall-public.cbhq.net/v1/code_change?repository=coinbase/geoengineer&ref=9eda8ddd995809b02ef2ae7796aca4ae501f8cfe)

**Comment by krobertson** [comment]
+1

**Comment by heimdall-asguard** [comment]
Review Error for krobertson@2018-01-29T22:34:09+00:00
User failed mfa authentication

**Comment by krobertson** [comment]
+1

**Comment by heimdall-asguard** [comment]
Reviews      | 4/1 | Message
------------ | ------------- | ----------
grahamjenson | 1 | +1 
chwevans | 1 | approved 
sds | 1 | approved 
krobertson | 1 | +1 



[More info](https://heimdall-public.cbhq.net/v1/code_change?repository=coinbase/geoengineer&ref=f14dd91f9491d3ebf139bf820b99c300b33c3d24)

**Comment by samnrubin** [comment]
+1

**Comment by heimdall-asguard** [comment]
Reviews      | 5/1 | Message
------------ | ------------- | ----------
grahamjenson | 1 | +1 
chwevans | 1 | approved 
sds | 1 | approved 
krobertson | 1 | +1 
samnrubin | 1 | +1 

Pull request is reviewed and can be merged to master!

[More info](https://heimdall-public.cbhq.net/v1/code_change?repository=coinbase/geoengineer&ref=f14dd91f9491d3ebf139bf820b99c300b33c3d24)

**Pull request merged**